src/webhook/dto/create-webhook-request.ts

```diff
@@ -1,23 +1,30 @@
 import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
-import { IsIn, IsNotEmpty, IsOptional, IsString } from 'class-validator';
+import { IsOptional, IsString, IsUrl, Length } from 'class-validator';
+import { NoDuplicateAndOtherTypes } from './custom-class-validators';
 import { WebhookEventType } from './webhook-event-types';
 
 export class CreateWebhookRequest {
-  @IsNotEmpty()
   @IsString()
+  @Length(4, 255, {
+    message: 'name should should have 4-255 characters',
+  })
   @ApiProperty()
   name: string;
 
-  @IsNotEmpty()
-  @IsString()
+  @NoDuplicateAndOtherTypes(
+    ['recipe_created', 'recipe_updated', 'recipe_deleted'],
+    {
+      message: `Invalid or duplicate event type detected in the array. Use these types: ${Object.values(
+        WebhookEventType,
+      )}`,
+    },
+  )
```

Comment:
hmm, I wouldn't do it like this.

First problem is that it's doing two things instead of one. It should either check only for a duplicated entries or check only the type, not both at once. For checking the type we can use:
```typescript
@IsArray()
@IsString({ each: true })
```

for the duplicates, I would handle it another way - I would just sanitize the input - that is I would remove the duplicated entries from the array. You can use `Transform` from `class-transformer` for sanitization purposes (https://github.com/typestack/class-transformer and https://www.npmjs.com/package/@nestjs/class-transformer).